internal/packer_registry/client.go

```diff
@@ -21,58 +24,75 @@ type ClientConfig struct {
 	ProjectID string
 }
 
-func (cfg ClientConfig) Validate() error {
-	if cfg.OrganizationID == "" {
-		return &ClientError{
+// NewClient returns an authenticated client to a HCP Packer Registry.
+// Client authentication requires the following environment variables be set HCP_CLIENT_ID and HCP_CLIENT_SECRET.
+// Upon error a HCPClientError will be returned.
+func NewClient() (*Client, error) {
+	if !env.HasHCPCredentials() {
+		return nil, &ClientError{
 			StatusCode: InvalidClientConfig,
-			Err:        errors.New(`no valid HCP Organization ID found, check that HCP_PACKER_REGISTRY is in the format "HCP_ORG_ID/HCP_PROJ_ID"`),
+			Err:        fmt.Errorf("the client authentication requires both HCP_CLIENT_ID and HCP_CLIENT_SECRET environment variables to be set"),
 		}
-
 	}
 
-	if cfg.ProjectID == "" {
-		return &ClientError{
+	cl, err := httpclient.New(httpclient.Config{})
+	if err != nil {
+		return nil, &ClientError{
 			StatusCode: InvalidClientConfig,
-			Err:        errors.New(`no valid HCP Project ID found, check that HCP_PACKER_REGISTRY is in the format "HCP_ORG_ID/HCP_PROJ_ID"`),
+			Err:        err,
 		}
 	}
 
-	return nil
-}
-
-// Client is an HCP client capable of making requests on behalf of a service principal
-type Client struct {
-	Config ClientConfig
-
-	Organization organization_service.ClientService
-	Project      project_service.ClientService
-	Packer       packerSvc.ClientService
-}
+	client := &Client{
+		Packer:       packerSvc.New(cl, nil),
+		Organization: organizationSvc.New(cl, nil),
+		Project:      projectSvc.New(cl, nil),
+	}
 
-// NewClient returns an authenticated client to a HCP Packer Registry.
-// Client authentication requires the following environment variables be set HCP_CLIENT_ID, HCP_CLIENT_SECRET, and HCP_PACKER_REGISTRY.
-// if not explicitly provided via a valid ClientConfig cfg.
-// Upon error a HCPClientError will be returned.
-func NewClient(cfg ClientConfig) (*Client, error) {
-	if err := cfg.Validate(); err != nil {
+	if err := client.loadOrganizationID(); err != nil {
 		return nil, &ClientError{
 			StatusCode: InvalidClientConfig,
 			Err:        err,
 		}
 	}
-
-	cl, err := httpclient.New(httpclient.Config{})
-	if err != nil {
+	if err := client.loadProjectID(); err != nil {
 		return nil, &ClientError{
 			StatusCode: InvalidClientConfig,
 			Err:        err,
 		}
 	}
 
-	svc := packerSvc.New(cl, nil)
-	return &Client{
-		Packer: svc,
-		Config: cfg,
-	}, nil
+	return client, nil
+}
+
+func (c *Client) loadOrganizationID() error {
+	// Get the organization ID.
+	listOrgParams := organizationSvc.NewOrganizationServiceListParams()
+	listOrgResp, err := c.Organization.OrganizationServiceList(listOrgParams, nil)
+	if err != nil {
+		return fmt.Errorf("unable to fetch organization list: %v", err)
+	}
+	orgLen := len(listOrgResp.Payload.Organizations)
```

Comment:
Can a service principle be tied to multiple orgs?

Reply:
No. They are created per org. Since we don't have any Org ID it is impossible to `GET` a single org, so it lists everything that the principal is able to see and this will always be one.